packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1060,45 +1070,102 @@ export function getSdkModelPropertyTypeBase(
   });
 }
 
-export function getSdkModelPropertyType(
+function updateMultiPartInfo(
   context: TCGCContext,
   type: ModelProperty,
-  operation?: Operation
-): [SdkModelPropertyType, readonly Diagnostic[]] {
+  base: SdkBodyModelPropertyType,
+  operation?: Operation,
+  httpOperationPart?: HttpOperationPart
+): [void, readonly Diagnostic[]] {
   const diagnostics = createDiagnosticCollector();
-  const base = diagnostics.pipe(getSdkModelPropertyTypeBase(context, type, operation));
+  const isBytesInput =
+    base.type.kind === "bytes" ||
+    (base.type.kind === "array" && base.type.valueType.kind === "bytes");
 
-  if (isSdkHttpParameter(context, type)) return getSdkHttpParameter(context, type, operation!);
-  // I'm a body model property
-  let operationIsMultipart = false;
-  if (operation) {
+  if (httpOperationPart) {
+    // body decorated with @multipartBody
+    const httpPart = getHttpPart(context.program, type.type);
+    base.multipartOptions = {
+      isFilePart:
+        isBytesInput ||
```

Comment:
```suggestion
        (!httpPart && isBytesInput) ||
```

We should not consider this `isBytesInput` if we have HttpPart.

Reply:
`httpOperationPart` doesn't have property to tell us whether this part is for File or not. we have to Judge it by ourselves.

Reply:
I am strongly against doing guesswork in the `HttpPart` logic.

If dev write `HttpPart<File>`, it is a file. If not, it is not.

One step back, if you really want to guess, it should be on `content-type`, not the type of the body.
Similar to "single", if `@body body: bytes`, it is still `application/json`, unless you specific e.g. `content-type=impage/png`.

Reply:
`HttpPart<bytes>` is a binary part and should be treated as such, File allows you to more finely control the filename and content type but they should been seen as the same over the wire. The goal is to still provide a good experience for the base types and not require always to go with the more precise one

The also produce the exact same openapi3 https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwpAc2VydmljZSh7CiAgdGl0bGU6ICJXaWRnZXQgU8YbIiwKfSkKbmFtZXNwYWNlIERlbW%2FHGzsKCm9wIHRlc3QoCiAgQG11bHRpcGFydEJvZHkgYm9keTogxFggIGE6IMRwUGFydDxieXRlcz47xRhiyxhGaWxlxRd9LAopOiB2b2lkOwo%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D

Reply:
So, `HttpPart<{@body body: bytes, @header contentType: "text/plain"}>` is also a file?

https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwpAc2VydmljZSh7CiAgdGl0bGU6ICJXaWRnZXQgU8YbIiwKfSkKbmFtZXNwYWNlIERlbW%2FHGzsKCm9wIHRlc3QoCiAgQG11bHRpcGFydEJvZHkgYm9keTogxFggIGE6IMRwUGFydDx7QMQaxx9ieXRlcywgQGhlYWRlciBjb250ZW505AClOiAidGV4dC9wbGFpbiJ9PjvFSWLLSUZpbGXFF30sCik6IHZvaWQ7Cg%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40typespec%2Fhttp%2Fall%22%5D%7D%7D

Reply:
I think weidong's concern make sense. It will be hard to judge whether it is file part since we may need judge it according to content-type and data type. So maybe `Only File stand for file part` is proper which is easy and explicit.
However, according to Typespec design doc, both `HttpPart<Bytes>` and `HttpPart<File>` stand for file part:

![image](https://github.com/Azure/typespec-azure/assets/70930885/59468ed8-4743-4522-8d7a-93c92fb8517e)

Add @timotheeguerin @bterlson for confirmation

Reply:
After discussion, we decide to treat `HttpPart<bytes>` as file part. Thanks all!